### PR TITLE
Ambient: support DNS backends with waypoint

### PR DESCRIPTION
**Pull request by howardjohn**
Today these are EDS clusters with 0 endpoints.

Added a test that excercises the full flow

### CODE REVIEW

File: pilot/pkg/networking/core/cluster_waypoint.go [hunk 1]
```diff
@@ -90,12 +91,22 @@ func (configgen *ConfigGeneratorImpl) buildWaypointInboundClusters(
...
-	localCluster := cb.buildCluster(clusterName, clusterType, nil,
-		model.TrafficDirectionInbound, &port, nil, nil)
+	discoveryType := convertResolution(cb.proxyType, svc)
+	var lbEndpoints []*endpoint.LocalityLbEndpoints
+	if discoveryType == cluster.Cluster_STRICT_DNS || discoveryType == cluster.Cluster_LOGICAL_DNS {
+		lbEndpoints = endpoints.NewCDSEndpointBuilder(
+			proxy,
+			cb.req.Push,
+			clusterName,
+			model.TrafficDirectionOutbound, "", svc.Hostname, port.Port,
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/cluster_waypoint.go:104]: Is this right changing from TrafficDirectionInbound to TrafficDirectionOutbound , and why

Comment (howardjohn) [reply pilot/pkg/networking/core/cluster_waypoint.go:104]: I think this doesn't actually get used. let me double check and switch to inbound

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/cluster_waypoint.go:104]: BTW the cluster name it is using inboundVIP

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/cluster_waypoint.go:104]: >clusterName := model.BuildSubsetKey(model.TrafficDirectionInboundVIP

This is a legacy issue

File: pilot/pkg/networking/core/cluster_waypoint.go [hunk 2]
```diff
@@ -90,12 +91,22 @@ func (configgen *ConfigGeneratorImpl) buildWaypointInboundClusters(
 }
 
 // `inbound-vip||hostname|port`. EDS routing to the internal listener for each pod in the VIP.
-func (cb *ClusterBuilder) buildWaypointInboundVIPCluster(svc *model.Service, port model.Port, subset string) *clusterWrapper {
+func (cb *ClusterBuilder) buildWaypointInboundVIPCluster(proxy *model.Proxy, svc *model.Service, port model.Port, subset string) *clusterWrapper {
 	clusterName := model.BuildSubsetKey(model.TrafficDirectionInboundVIP, subset, svc.Hostname, port.Port)
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pilot/pkg/networking/core/cluster_waypoint.go:95]: done

**Review by stevenctl — APPROVED** [review]

**Pull request merged**